src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -365,10 +365,33 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
         }
         delegateCheckpoints <- processAccountsForBlocks(accountUpdates, rollsData) // should this fail, we still recover data from the checkpoint
         _ <- processDelegatesForBlocks(delegateCheckpoints) // same as above
+        _ <- processBlocksForGovernance(blocks)
       } yield results.size
 
     }
 
+    /** Processes blocks and fetches needed data to produce Governance rows*/
+    def processBlocksForGovernance(blocks: List[Block]): Future[Unit] = {
+      import DatabaseConversions._
+      import tech.cryptonomic.conseil.util.Conversion.Syntax._
+
+      for {
+        proposals: List[(BlockHash, Option[ProtocolId])] <- tezosNodeOperator.getProposals(blocks.map(_.data))
+        blockHashesWithProposals = proposals.filter(_._2.isDefined).map(_._1)
+        blocksWithProposals = blocks.filter(blockData => blockHashesWithProposals.contains(blockData.data.hash))
+        (listings, ballots, ballotCounts) <- {
+          if (blocksWithProposals.nonEmpty) {
+            tezosNodeOperator.getVotes(blocksWithProposals)
+          } else {
+            Future.successful((List.empty, List.empty, List.empty))
+          }
+        }
+        governanceRows = (blocksWithProposals.map(_.data), proposals, listings, ballots, ballotCounts)
```

Comment:
My opinion on this is that it's more clear to define conversions for each block's bundled data (which might be a custom class made just for this use).
The we map directly with the convertTo function.

It might look more complicated seen from the call site (here), but it simplifies the logic of a conversion function which is too custom tailored to this specific use to be easily refactored/updated later on.

My position on this is that, while we want to simplify how the code is called to make intent clear, we should nonetheless keep the single pieces of logic consistent with an understandable design choice, and easy to compose in a reasonable way.
The idea of a Conversion function is that you have a data type that can be converted to another or more than one. The input data type might be a composition, out of necessity (because we might not have all the required values in  a single object) yet making the input arbitrarily complex for ease of use is going overboard in my opinion.

As a guideline I would strive as much as possible to reduce the input elements to the minimum required to convert to some other data element, eventually creating temporary input class to aggregate multiple pieces of information when they're more than a few.

Reply:
ok, changed